### PR TITLE
Add the possibility to filter loaded actors by tags.

### PR DESCRIPTION
**Pull request by pcahyna**
This is needed to be able to choose actors which have all given tags. The load() function takes a list of tags, but loads actors which have any of them. If we want to load actors which have all of them, we must load them using one of tags and then filter them using the new actor_names_by_tags function.

The motivation is that we will eventually have a large number of actors, shared among different projects (Leapp, preupgrade-assistant, ...). If we tag the preupgrade assistant ones with e.g. a "p-a" tag and use various other tags, e.g. "readonly" for actors that do not modify the system, the preupgrade-assistant application might be interested in loading only those actors that are intended for preupgrade-assistant AND are readonly. By using 
```
load(..., tags=("p-a", "readonly"))
```
it would get those that are intended for preupgrade-assistant OR readonly, which is not what it would need.

More urgently, this is needed to start composing actors using leappwf. We need to tag actors that are ready to be used with leappwf. E.g. the check-target implementation using leappwf and implicit workflow construction by dependency resolution will need to use actors tagged with "check_target", but excluding the check_target_group one. This can be done by tagging those that we want by a "depsolver" tag, but currently we don't have any means to select those that have both the "depsolver" and "check_target" tags. This PR fills this gap.

### CODE REVIEW

File: snactor/registry/actors.py [hunk 1]
```diff
@@ -19,6 +19,13 @@ def must_get_actor(actor):
     return _instantiate_actor(_REGISTERED_ACTORS[actor])
 
 
+def actor_names_by_tags(tags, names=()):
+    if names == ():
```
Comment (vinzenz) [inline snactor/registry/actors.py:23]:
```python
if names:
```

File: snactor/registry/actors.py [hunk 2]
```diff
@@ -19,6 +19,13 @@ def must_get_actor(actor):
...
+        return [name for name, a  in _REGISTERED_ACTORS.items() if a[0].tags.intersection(tags)]
```
Comment (vinzenz) [inline snactor/registry/actors.py]: intersection isn't checking for subsets so if you want that then do check for subsets  
```python
return [name for name, a  in _REGISTERED_ACTORS.items() if a[0].tags.issubset(tags)]
```

**Review by vinzenz — CHANGES_REQUESTED** [review]

File: snactor/registry/actors.py [hunk 3]
```diff
@@ -19,6 +19,13 @@ def must_get_actor(actor):
     return _instantiate_actor(_REGISTERED_ACTORS[actor])
 
 
+def actor_names_by_tags(tags, names=()):
+    if names == ():
+        return [name for name, a in _REGISTERED_ACTORS.items() if a[0].tags.intersection(tags)]
```
Comment (vinzenz) [inline snactor/registry/actors.py:24]: intersection isn't checking for subsets so if you want that then do check for subsets  
```python
return [name for name, a  in _REGISTERED_ACTORS.items() if a[0].tags.issubset(tags)]
```

Comment (pcahyna) [reply snactor/registry/actors.py:24]: I want to select actors which have any of the tags, not all of them.
When one wants actors which have all tags, successive filters would be applied.
(the goal is to behave just like the tag list of load())

File: snactor/registry/actors.py (hunk 1 above)
Comment (pcahyna) [reply snactor/registry/actors.py:23]: actually this should be done differently:
```python
def actor_names_by_tags(tags, names=None):
    if names is None:
```
the reason is that passing an empty list should be different than not passing anything. The former means that we want no actors, the latter that we select from all actors.

**Comment by vinzenz** [comment]
99f0127715b4b15714beae9254c7529a4ff6ae69 was merged as 71e1c78a6dfdaf5b2d20516ef7b2217d3241afc6

**Pull request closed without merging**